### PR TITLE
move config and session files to $XDG_CONFIG_DIR

### PR DESCRIPTION
**Pull request by treeshateorcs**
this is probably ugly. my first pull request in rust

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -29,9 +30,13 @@ impl fmt::Display for ConfigError {
 }
 
 pub fn load_config() -> Result<Config, ConfigError> {
-    let buffer = fs::read_to_string("config.toml")
+    let mut path = dirs::config_dir().unwrap();
+    path.push("rescrobbled");
+    fs::create_dir_all(&path).expect("could not create config dir");
+    path.push("config.toml");
+    let buffer = fs::read_to_string(path.into_os_string().into_string().unwrap())
```
Comment (InputUsername) [inline src/config.rs]: Same thing here, simply passing `&path` works.

**Comment by treeshateorcs** [comment]
done! please check, and sorry for the noob mistakes

**Comment by InputUsername** [comment]
Perfect, thank you! And no worries, I'm a rust beginner myself.

Are you okay with the license by the way?

**Comment by treeshateorcs** [comment]
I would prefer GPL, but it's your call

MIT is too permissive in my opinion. evil people not willing to contribute to open source may use your software
https://drewdevault.com/2019/06/13/My-journey-from-MIT-to-GPL.html

**Comment by treeshateorcs** [comment]
also i wonder why this program had 0 stars. there's basically only one alternative and it's glitchy

I'm working on ListenBrainz API, so this package can scrobble to ListenBrainz too

**Comment by InputUsername** [comment]
Generally the rust community has preferred MIT or Apache (or dual license) so that's mostly why I picked MIT, I can see where you're coming from though. Since it's not a library I think GPL is fine.

As for why the program has no stars, I've never announced it anywhere, it was mostly a project for personal use but additional contributions are always welcome!

**Comment by treeshateorcs** [comment]
please take a look at 
https://github.com/treeshateorcs/listenbrainz-rust

I spent ~15 hours on those 100 SLOC :1st_place_medal:

**Pull request merged**